### PR TITLE
Rescue Master Branch PR: MNT Use SiteTree instead of Page in tests

### PR DESCRIPTION
**Pull request by GuySartorelli**
Rescues https://github.com/silverstripe/silverstripe-cms/pull/2160/

Note that I've swapped out `::create()` syntax for using the `new` keyword, which is preferred for all tests that don't explicitly need to rely on `Injector`.

## Parent issue
- https://github.com/silverstripe/silverstripe-framework/issues/10350

### CODE REVIEW

File: tests/php/Model/SiteTreeTest.php [hunk 1]
```diff
@@ -470,46 +479,47 @@ public function testGetByLink()
         );
     }
 
-    public function testGetByLinkAbsolute()
```
Comment (emteknetnz) [inline tests/php/Model/SiteTreeTest.php:473]: Why is this being removed?

File: tests/php/Model/SiteTreeTest.php [hunk 2]
```diff
@@ -1634,36 +1672,13 @@ public function testLinkExtension()
     }
 
     /**
-     * Test that the controller name for a SiteTree instance can be gathered by appending "Controller" to the SiteTree
+     * Test that the controller name for a Page instance can be gathered by appending "Controller" to the Page
      * class name in a PSR-2 compliant manner.
      */
     public function testGetControllerName()
     {
-        $class = new Page;
```
Comment (emteknetnz) [inline tests/php/Model/SiteTreeTest.php:1642]: Why is all of this being removed?

**Review by emteknetnz — CHANGES_REQUESTED** [review]
Seems fine, not sure why a couple of blocks of test code is being removed though

File: tests/php/Model/SiteTreeTest.php (hunk 2 above)
Comment (ChloeSartorelli) [reply tests/php/Model/SiteTreeTest.php:1642]: Looks like a mistake when I was resolving conflicts. Will add back in.

File: tests/php/Model/SiteTreeTest.php (hunk 1 above)
Comment (ChloeSartorelli) [reply tests/php/Model/SiteTreeTest.php:473]: Looks like a mistake when I was resolving conflicts. Will add back in.

File: tests/php/Model/SiteTreeTest.php (hunk 2 above)
Comment (ChloeSartorelli) [reply tests/php/Model/SiteTreeTest.php:1642]: Done

File: tests/php/Model/SiteTreeTest.php (hunk 1 above)
Comment (ChloeSartorelli) [reply tests/php/Model/SiteTreeTest.php:473]: Done

**Review by emteknetnz — APPROVED** [review]

**Pull request merged**